scroll-animations-1/Overview.bs

```diff
@@ -789,28 +770,72 @@ follows:
+        add the resulted value into <var>effective scroll offsets</var>.
+1.  If {{ScrollTimeline/scrollOffsets}} has one element
+    1.  Run the procedure to [=resolve a scroll timeline offset=] for 'auto'
+        with the <var>is first</var> flag set to <var>first offset</var> and
+        add the resulted value into <var>effective scroll offsets</var>.
+    1.  Set <var>first offset</var> to false.
+1.  For each <var>scroll offset</var> in the list of
+    {{ScrollTimeline/scrollOffsets}}, perform the following steps:
+    1.  Let <var>effective offset</var> be the result of applying the procedure
+        to [=resolve a scroll timeline offset=] for <var>scroll offset</var>
+        with the <var>is first</var> flag set to <var>first offset</var>.
+    1.  If <var>effective offset</var> is null, the
+        [=effective scroll offsets=] is empty and abort the remaining steps.
+    1.  Add <var>effective offset</var> into
+        <var>effective scroll offsets</var>.
+    1.  Set <var>first offset</var> to false.
+1.  Return <var>effective scroll offsets</var>.
+
+### The progress of a {{ScrollTimeline}} ### {#progress-calculation-algorithm}
+The <dfn>scroll timeline progress</dfn> is calculated by applying the procedure
+to <dfn>calculate scroll timeline progress</dfn> given |offset| as follows:
+
+1.  Let <var>scroll offsets</var> be the result of applying the procedure to
+    [=resolve scroll timeline offsets=] for {{scrollOffsets}}.
+1.  Let <var>offset index</var> correspond to the position of the last offset in
+    <var>scroll offsets</var> whose value is less than or equal to
+    <var>offset</var> and the value at the following position greater than
+    <var>offset</var.
+1.  Let <var>start offset</var> be the offset value at position
+    <var>offset index</var> in <var>scroll offsets</var>.
+1.  Let <var>end offset</var> be the value of next offset in <var>scroll
+    offsets</var> after <var>start offset</var>.
+1.  Let <var>size</var> be the number of offsets in <var>scroll offsets</var>.
+1.  Let <var>offset weight</var> be the result of evaluating
+    <code> 1 / (<var>size</var> - 1)</code>.
+1.  Let <var>interval progress</var> be the result of evaluating
+    <code>(<var>offset</var> - <var>start offset</var>) /
+    (<var>end offset</var> - <var>start offset</var>)</code>.
+1.  Return the result of evaluating
+    <code>(<var>offset index</var> + <var>interval progress</var>) &times;
+    <var>offset weight</var></code>.
+
+Note: This procedure guarantees that given <var>offset</var> belongs to
+scroll range within first and last offsets of <var>scroll offsets</var>.
+This is because the case when <var>offset</var> falls outside of the scroll
+range is handled before calling this procedure, as part of [=current time=]
+calculation.
+
+Note: The rationale behind choosing last matching offset in the array is to
+be consistent with CSS property overrides.
```

Comment:
I think we should link to the rule for finding which KeyframeRule applies in a comma separated list of keyframes with overlap as justification: https://drafts.csswg.org/css-animations-1/#interface-csskeyframesrule-findrule

Reply:
Done.